### ISSUE TITLE
[MemberNotNull] annotations is not processed on implicit calls from collection initializers

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#43561]
**Version Used**: 

```
commit a1b867054dff97e5e5f44c7ad742c8ffb740b2eb
Merge: 8f4fdad99ae 6bf71e1a37d
Author: msftbot[bot] <48340428+msftbot[bot]@users.noreply.github.com>
Date:   Wed Apr 22 09:50:57 2020 +0000

    Merge pull request #43554 from CyrusNajmabadi/renameOOP4

    Move the 'finding' portoin of rename out of process.
```

**Steps to Reproduce**:

Compile the following code
```cs

using System.Collections;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;

class C : IEnumerable
{
  public IList<string>? collection;

  [MemberNotNull("collection")]
  public void Add(string s) => (collection ??= new List<string>()).Add(s);
  public IEnumerator GetEnumerator() => throw null!;

  void Test()
  {
    var c = new C { "" };
    c.collection.ToString();
  }
}
```

**Expected Behavior**:
No warnings. The `Add` call from collection initializer guarantees that `collection` field is not null and it's reflected in its annotation

**Actual Behavior**:
Warning `CS8602: Dereference of a possibly null reference` is reported for `c.collection`

**Notes**
If you rewrite the code to call `Add` explicitly i.e.
```
var c = new C();
c.Add("");
```
the warning goes away
